dataflow/src/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -142,7 +148,16 @@ protected S runAnalysisFor(Node node, boolean before) {
         if (transferInput == null) {
             return null;
         }
-        return runAnalysisFor(node, before, transferInput);
+        IdentityHashMap<Node, TransferResult<A, S>> cache = this.cache.get(transferInput);
+        if (cache == null) {
+            this.cache.put(transferInput, cache = new IdentityHashMap<>());
+        }
+        return runAnalysisFor(node, before, transferInput, cache);
+    }
+
+    public static <A extends AbstractValue<A>, S extends Store<S>> S runAnalysisFor(
+            Node node, boolean before, TransferInput<A, S> transferInput) {
+        return runAnalysisFor(node, before, transferInput, null);
```

Comment:
How many places call this method? Can we instead pass the right cache in those locations?